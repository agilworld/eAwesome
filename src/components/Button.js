import React from 'react';
import { TouchableHighlight, StyleSheet, Text } from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 0,
  },
  buttonPlain:{
    paddingRight:10
  },
  buttonContained: {
    height: 40,
    backgroundColor:"#E04081",
    color:'#fff',
    borderRadius: 4,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop:10,
  },
  textPlain:{
    color:'#333'
  },   
  textContained:{
      color:'#fff'
  }
});

const Button = props => {
  const {
    label,
    onPress,
    variant,
    ...restProps
  } = props

  let buttonStyle, textStyle
  let text = (label == undefined) ? 'Button' : label
  if( variant == 'contained' ) {
    buttonStyle = styles.buttonContained
    textStyle = styles.textContained
  } else {
    buttonStyle = styles.buttonPlain
    textStyle = styles.textPlain
  }

  return (
    <TouchableHighlight 
        testID={"reusedButton"}
        style={buttonStyle}
        underlayColor={'#fefefe'}
        onPress={onPress}
        {...restProps}
    >
        <Text style={textStyle}>{text}</Text>
    </TouchableHighlight>
  );
};

export default Button;
