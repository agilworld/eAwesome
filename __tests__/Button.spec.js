import React from 'react';
import {render,fireEvent} from 'react-native-testing-library';
import Button from '../src/components/Button';

describe('<Button />', () => {
    
    describe('test label & press', () => {
        it('should render correctly', () => {
            const buttonHandler = jest.fn();
            const {queryByText, getByTestId} = render(<Button label="Button" onPress={buttonHandler}/>)
            expect(queryByText('Button')).not.toBeNull()
            fireEvent.press(getByTestId('reusedButton'))
        })
    })

    describe('Good case - Pass some props', () => {
        const props = {
            label:'Sign In',
            variant:'contained',
            onPress:() => {}
        }
        it('should render correctly', () => {
            const wrapper = render(<Button {...props} />)
            expect(wrapper.toJSON()).toMatchSnapshot()
        })

        it('passing test label', () => {
            const {queryByText} = render(<Button label="Sign Out"/>)
            expect(queryByText('Sign Out')).toBeTruthy()
        })

        it('passing onpress button', () => {
            const buttonHandler = jest.fn()
            const {getByTestId} = render(<Button onPress={buttonHandler} />)

            fireEvent.press(getByTestId('reusedButton'))
        })
    })

    describe('Bad case - No pass any props', () => {
        it('should render correctly', () => {
            const wrapper = render(<Button />)
            expect(wrapper.toJSON()).toMatchSnapshot()
        })

        it('will show label Button', () => {
            const {queryByText} = render(<Button />)
            expect(queryByText('Button')).toBeTruthy()
        })

    })
});