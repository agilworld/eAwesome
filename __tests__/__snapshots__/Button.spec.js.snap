// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Bad case - No pass any props should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "paddingRight": 10,
    }
  }
  testID="reusedButton"
>
  <Text
    style={
      Object {
        "color": "#333",
      }
    }
  >
    Button
  </Text>
</View>
`;

exports[`<Button /> Good case - Pass some props should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E04081",
      "borderRadius": 4,
      "color": "#fff",
      "height": 40,
      "justifyContent": "center",
      "marginTop": 10,
    }
  }
  testID="reusedButton"
>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Sign In
  </Text>
</View>
`;
